pkg/apis/duck/v1alpha1/pubsub_defaults.go

```diff
@@ -17,13 +17,25 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"context"
+
+	"github.com/google/knative-gcp/pkg/apis/configs/gcpauth"
+
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/equality"
+	"knative.dev/pkg/apis"
 )
 
-func (s *PubSubSpec) SetPubSubDefaults() {
-	if s.GoogleServiceAccount == "" && s.ServiceAccountName == "" &&
-		(s.Secret == nil || equality.Semantic.DeepEqual(s.Secret, &corev1.SecretKeySelector{})) {
-		s.Secret = DefaultGoogleCloudSecretSelector()
+func (s *PubSubSpec) SetPubSubDefaults(ctx context.Context) {
+	ad := gcpauth.FromContextOrDefaults(ctx).AuthorizationDefaults
```

Comment:
nit: shall this be a method that returns (ad, err)? and then we check for err == nil ?
I'm ok if not

Reply:
The pattern I have seen is that it returns nil, but I could see it being better to return an err instead. In the future I would like to move to `FromContext` when I handle the TODO a few lines down.